### DISCUSSION TITLE
Help With Syncing Past Workouts

### DISCUSSION THREAD

**Discussion by alanbk266** [philosowaffle/peloton-to-garmin#159] [Q&A]
I got and set up my Garmin device at the end of July.  I've had the Peloton for two years.  Installed V2 on Windows, set  NumWorkoutsToDownload to 1800.  I have about 1770.  When I ran the program, it sync'd everything from the end of July on, certainly not 1800. maybe a little over 100.  When I tried to re-run it said...
[18:45:43 WRN] App.PythonAndGuploadInstalledLocally setting is deprecated and will be removed in a future release. Please swith to using Garmin.UploadStrategy config.
[18:45:44 INF] No files to convert in download directory. Nothing to do.
[18:45:44 INF] No upload directory found. Nothing to do.
[18:45:44 INF] Done.  

Config file...
{
  // Config Documentation: https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration
  "App": {
    "OutputDirectory": "./output",
    "WorkingDirectory": "./working",
    "SyncHistoryDbPath": "./output/syncHistory.json",
    "EnablePolling": false,
    "PollingIntervalSeconds": 86400,
    "PythonAndGUploadInstalled": false,
	"CloseWindowOnFinish": false
  },

  "Format": {
    "Fit": true,
    "Json": false,
    "Tcx": false,
    "SaveLocalCopy": true,
    "IncludeTimeInHRZones": false,
    "IncludeTimeInPowerZones": false,
    "DeviceInfoPath": "./deviceInfo.xml"
  },

  "Peloton": {
    "Email": " 
    "Password":  
    "NumWorkoutsToDownload": 1800,
    "ExcludeWorkoutTypes": [ "meditation" ]
  },

  "Garmin": {
    "Email":  ",
    "Password": 
    "Upload": true,
    "FormatToUpload": "fit",
    "UploadStrategy": 2
  },

  "Observability": {

    "Prometheus": {
      "Enabled": false,
      "Port": 4000
    },

    "Jaeger": {
      "Enabled": false,
      "AgentHost": "localhost",
      "AgentPort": 6831
    },

    "Serilog": {
      "Using": [ "Serilog.Sinks.Console", "Serilog.Sinks.File" ],
      "MinimumLevel": "Information",
      "WriteTo": [
        { "Name": "Console" },
        {
          "Name": "File",
          "Args": {
            "path": "./output/log.txt",
            "rollingInterval": "Day",
            "retainedFileCountLimit": 7
          }
        }
      ]
    }
  }
}

**Comment by philosowaffle** [comment]
I've got a hunch on what the problem is. Will try to investigate more and get a fix in this week.

**Reply by alanbk266** [reply]
Thanks!

**Reply by philosowaffle** [reply]
@alanbk266 worked on this today and believe I have it fixed (#160).  You can pull the latest version of the `master` branch or use the windows exe found [here](https://github.com/philosowaffle/peloton-to-garmin/actions/runs/1155978038)

I only had about 270 workouts to test with, so let me know if it successfully works for your 1500+ (congrats on that number btw!).

**Comment by alanbk266** [comment]
Thank you very much I really appreciate it! I will try later today! Just
curious what was the problem? I work for a medical record software company.
I’m not a developer. I manage the SRE team.

**Reply by philosowaffle** [reply]
I completely forgot to add support for paging when fetching workouts from Peloton. So no matter how many workouts you asked for, it only ever fetched the first page of data, and Peloton only returns 100 items per page.  So the fix was to continue iterating through the pages of data so we could grab older and older workouts.

**Comment by alanbk266** [comment]
Awesome, thanks!

Hey another question for you I’m actually looking to either build myself or
have a developer do it a special app I want to do in  peloton, probably it
would be relatively simple for you since you already know how to make the
API calls to peloton. Of course pay you for your development time, or
getting me up to speed on the API calls to peloton.So if you’re interested
let me know, my email address is ***@***.*** If you could email me
and we will communicate that way.  I really don’t want to put this idea out
in public.
, and I know this is gonna go on to the forum.


Thanks again and I’m about to try and sync my rides.

**Comment by alanbk266** [comment]
The latest version does not seem to recognize the control file.  I'd rather
now use cmd line parms, or evn variables.  Thoughts?

Thanks,

Akan
